python/cudnn/sdpa/graph_analyzer.py

```
@@ -749,3 +758,164 @@ def tensor_desc_from_ir(t: Any, name: str = "") -> "TensorDesc":
+        out["block_mask"] = ops.block_mask
+    if ops.alibi:
+        out["alibi"] = True
+    return out
+
+
+def to_bshd_physical(t: "torch.Tensor") -> "torch.Tensor":
+    """BSHD-physical (stride order 3,1,2,0) copy of a rank-4 BHSD-logical
+    tensor; zero-copy when the buffer already is.  Delivers the dense_flex
+    layout relaxation for lowerings whose adapters require this order
+    (mirrors the DSL executor's canonical-buffer gather)."""
+    b, h, s_, d = t.shape
```

Ruff (0.16.1)

[warning] Unpacked variable `b` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)